src/components/case-details-card.jsx

```diff
@@ -36,6 +36,118 @@ const JustificationBox = ({ onChange, justification }) => {
 };
 
+const getVotingStatus = (disputePeriod, votesData, hiddenVotes) => {
+  if (disputePeriod === "0") {
+    return "Waiting for evidence.";
+  } else if (disputePeriod === "1") {
+    return !votesData.committed
+      ? "You did not commit your vote yet."
+      : "You committed your vote. You will be able to reveal your vote when the period ends.";
+  } else if (hiddenVotes) {
+    return votesData.committed
+      ? "You did not reveal your vote yet."
+      : "You did not commit a vote in the previous period. You cannot vote anymore.";
+  } else {
+    return "You did not cast a vote.";
+  }
+};
+
+const VoteOptions = ({ metaEvidence, votesData, complexRuling, setComplexRuling, onVoteClick, disabledDate }) => {
+  const isSingleSelect = metaEvidence.rulingOptions.type === "single-select";
+  const isMultipleSelect = metaEvidence.rulingOptions.type === "multiple-select";
+  const isDateTime = metaEvidence.rulingOptions.type === "datetime";
+
+  let inputComponent;
+
+  if (isMultipleSelect) {
+    inputComponent = (
+      <div style={{ paddingTop: "1rem" }}>
+        <Checkbox.Group
+          disabled={!votesData.canVote}
+          name="ruling"
+          onChange={setComplexRuling}
+          options={metaEvidence.rulingOptions.titles?.slice(0, 255)}
+          value={complexRuling}
+        />
+      </div>
+    );
+  } else if (isDateTime) {
+    inputComponent = (
+      <DatePicker
+        disabled={!votesData.canVote}
+        disabledDate={disabledDate}
+        onChange={setComplexRuling}
+        size="large"
+        showTime
+        value={complexRuling}
+      />
+    );
+  } else if (isSingleSelect) {
+    inputComponent = metaEvidence.rulingOptions.titles?.slice(0, 2 ** 256 - 1).map((title, index) => (
+      <StyledButton
+        disabled={!votesData.canVote}
+        id={index + 1}
+        key={title}
+        onClick={onVoteClick}
+        size="large"
+        type="primary"
+      >
+        {title}
+      </StyledButton>
+    ));
+  } else {
+    inputComponent = (
+      <InputNumber
+        disabled={!votesData.canVote}
+        max={Number(
+          realitioLibQuestionFormatter
+            .maxNumber({
+              decimals: metaEvidence.rulingOptions.precision,
+              type: metaEvidence.rulingOptions.type,
+            })
+            .minus(1)
+        )}
+        min={Number(
+          realitioLibQuestionFormatter.minNumber({
+            decimals: metaEvidence.rulingOptions.precision,
+            type: metaEvidence.rulingOptions.type,
+          })
+        )}
+        onChange={setComplexRuling}
+        precision={metaEvidence.rulingOptions.precision}
+        size="large"
+        value={complexRuling}
+      />
+    );
+  }
+
+  return (
+    <StyledButtonsDiv>
+      {inputComponent}
+      {!isSingleSelect && (
+        <StyledButton disabled={!votesData.canVote || !complexRuling} onClick={onVoteClick} size="large" type="primary">
+          Submit
+        </StyledButton>
+      )}
```

Comment:
I think it's more clear with this syntax:
```suggestion
      {isSingleSelect
        ? null
        : (
          <StyledButton disabled={!votesData.canVote || !complexRuling} onClick={onVoteClick} size="large" type="primary">
            Submit
          </StyledButton>
        )
      }
```